python/CMakeLists.txt

```diff
@@ -58,6 +58,9 @@ endif()
 
 # Top level cmake dir
 if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
+  option(PYARROW_BUILD_ARROW_GPU
+    "Build the PyArrow GPU support"
+    OFF)
```

Comment:
Maybe rename this to `PYARROW_BUILD_ARROW_CUDA`. At some point we need to resolve the issue that `libarrow_gpu` is CUDA-specific, don't want to propagate the issue here if possible

Reply:
Renamed `PYARROW_BUILD_ARROW_GPU` to `PYARROW_BUILD_CUDA`.
`libarrow_gpu` should probably also renamed to `libarrow_cuda`. I'll leave it to the second round for now..,